src/main/scala/mrpowers/jodie/DeltaHelpers.scala

```diff
@@ -240,4 +242,47 @@ object DeltaHelpers {
                       cols: List[String],
                       newColName: String
                     ): DataFrame = withMD5Columns(deltaTable.toDF, cols, newColName)
+  
+
+  /**
+   * this function allows to delete rows from a DeltaTable from rows of a dataframe with join of one or more columns and the comparison operator like ("=",">","!=",...)
+   * and return number of deleted rows
+   * @param deltaTable
+   *   : delta table object.
+   * @param dataFrame
+   *   : dataframe source
+   * @param attrColNameOper
+   *   : Map[("column Name", "Operator")]
+   */
+
+  def deleteFromAnotherDataframe
+                  (
+                  targetTableDelta: io.delta.tables.DeltaTable,
+                  sourceDF: org.apache.spark.sql.DataFrame,
+                  attrColNameOper: Map[String,String]
+                  ) : Int = 
+  {
+
+    var returnValue  : Int = 0
+
+    Try{
```

Comment:
The `Try,Success,Failure` pattern should be used to return a `Try` not a concrete type, for example 

```scala
def deleteFromAnotherDataframe(
        targetTableDelta: io.delta.tables.DeltaTable,
        sourceDF: org.apache.spark.sql.DataFrame,
        attrColNameOper: Map[String,String]) : Try[Int] = {
        previous code...
        previous code...
       Try{
           your code here....
       }
}
```

And the users consuming your api are the ones that will perform the Sucess or failure pattern, like this

```scala
deleteFromAnotherDataframe(x,y,z) match {
    case Success(i) => println("do success actions here.")
    case Failure(s) => println("handle errors here.")
}
```

I would advice in this case that insted of `Try,Success,Failure` use `try,catch` and handle specific errors like invalid columns or invalid operators with a JodieValidationError message.